rclpy/rclpy/node.py

```diff
@@ -554,12 +567,10 @@ def destroy_publisher(self, publisher: Publisher) -> bool:
 
         :return: ``True`` if successful, ``False`` otherwise.
         """
-        for pub in self.publishers:
-            if pub.publisher_handle == publisher.publisher_handle:
-                with self.handle as capsule:
-                    _rclpy.rclpy_destroy_node_entity(pub.publisher_handle, capsule)
-                self.publishers.remove(pub)
-                return True
+        if publisher in self.publishers:
+            self.publishers.remove(publisher)
+            publisher.destroy()
```

Comment:
For all of these destroy calls, do we need to check for the `InvalidHandle` exception?  Is that a case that can happen?

Reply:
It is. Maybe this should catch it and return `False`?

Reply:
Yeah, that seems reasonable.

Reply:
`destroy_<entity>()` catch `InvalidHandle` and return `False` in e630404